solidity/contracts/StaticOracle.sol

```diff
@@ -160,13 +199,18 @@ contract StaticOracle is IStaticOracle {
     address _baseToken,
     address _quoteToken,
     address[] memory _pools,
-    uint32 _period
+    uint32 _period,
+    uint32 _offset
   ) internal view returns (uint256 _quoteAmount) {
     require(_pools.length > 0, 'No defined pools');
     OracleLibrary.WeightedTickData[] memory _tickData = new OracleLibrary.WeightedTickData[](_pools.length);
+
+    // If period is 0 but offsetted, we force a 1s twap calculation
+    _period = _period == 0 && _offset > 0 ? 1 : _period;
+
     for (uint256 i; i < _pools.length; i++) {
       (_tickData[i].tick, _tickData[i].weight) = _period > 0
-        ? OracleLibrary.consult(_pools[i], _period)
+        ? OracleLibraryPlus.consultOffsetted(_pools[i], _period, _offset)
         : OracleLibrary.getBlockStartingTickAndLiquidity(_pools[i]);
```

Comment:
@0xsambugs wdyt about this implementation?
since `getBlockStartingTickAndLiquidity` is unavailable for old timestamps, i've decided to just query a 1s offsetted twap, is ofc not recommended, but open for ppl to do it

Reply:
I think it could lead to unwanted situations. I'd rather we revert with a clear message, so the user takes notice and they change it to 1 second by themselves. Otherwise, we could be doing something they don't want to do

Reply:
roger, i believe quoting a 1s period is the same as asking the current tick and liquidity (bc it extrapolates by 1s, and then divides by 1s), but added a revert if `_offset != 0 && _period ==0`
feel free to comment on the string error 😇

Reply:
added also a test for such scenario under `test:unit`